### PR TITLE
Add maveriknight's custom item

### PR DESCRIPTION
**Pull request by HeyBanditoz**
[Forum post in question](https://baystation12.net/forums/threads/old-camera-kristen-osterweis-maveriknight.5432/)

### CODE REVIEW

**Comment by HeyBanditoz** [comment]
Is there any way to test these changes?

File: config/custom_items.txt [hunk 1]
```diff
@@ -778,4 +778,15 @@ item_path: /obj/item/clothing/accessory/storage/white_vest
...
+
+{
+ckey: maveriknight
+character_name: Kristen Osterweis
+item_path: /obj/item/device/camera
+item_name: Old Camera
+item_icon: osterweis_oldcamera
+item_desc: An old Ward-Takahashi brand personal camera, with a built-in photo developer. Its edges look worn from use. The silver paint on the bottom has been worn away to reveal simple grey plastic beneath, but it looks like the camera still works.
+req_titles: Passenger
+req_access: 1
```
Comment (mustafakalash) [inline config/custom_items.txt]: You don't want the `req_access` line.

**Review by mustafakalash — CHANGES_REQUESTED** [review]
This may require changes to https://github.com/Baystation12/Baystation12/blob/dev/code/modules/paperwork/photography.dm#L164-L171 on the main repo to properly work with the custom icon.

Change `attack_self` to
```dm
on = !on
update_icon()
```
add a `/obj/item/device/camera/update_icon`:
```dm
var/datum/extension/base_icon_state/bis = get_extension(src, /datum/extension/base_icon_state)
if(on)
    icon_state = "[bis.base_icon_state]"
else
    icon_state = "[bis.base_icon_state]_off"
```
Then, add `/obj/item/device/camera/New`:
```dm
set_extension(src, /datum/extension/base_icon_state, /datum/extension/base_icon_state, icon_state)
update_icon()
```
Finally, you're going to want to get an `osterweis_oldcamera_off` with a red light.

Let me know if you have any questions.

**Comment by HeyBanditoz** [comment]
Oh God the spam. Anyway, I think this is ready @mustafakalash if you'd like to check it.

**Review by mustafakalash — APPROVED** [review]
Dependent on Baystation12/Baystation12#19129

**Pull request merged**